packages/assets-controllers/src/TokenDetectionController.ts

```diff
@@ -588,56 +612,68 @@ export class TokenDetectionController extends StaticIntervalPollingController<To
 
     const addressAgainstWhichToDetect =
       selectedAddress ?? this.#getSelectedAddress();
-    const { chainId, networkClientId: selectedNetworkClientId } =
-      this.#getCorrectChainIdAndNetworkClientId(networkClientId);
-    const chainIdAgainstWhichToDetect = chainId;
-    const networkClientIdAgainstWhichToDetect = selectedNetworkClientId;
 
-    if (!isTokenDetectionSupportedForNetwork(chainIdAgainstWhichToDetect)) {
-      return;
-    }
-    if (
-      !this.#isDetectionEnabledFromPreferences &&
-      chainIdAgainstWhichToDetect !== ChainId.mainnet
-    ) {
+    // Will be an array of objects in the form [{chainId : "0x1", networkClientId: "8e8b86f2-f949-4f29-90bc-f7409f0832d5"}]
+    const clientNetworksIdByChainId =
+      this.#getCorrectNetworkClientIdByChainId(chainIds);
+
+    if (!clientNetworksIdByChainId) {
       return;
     }
-    const isTokenDetectionInactiveInMainnet =
-      !this.#isDetectionEnabledFromPreferences &&
-      chainIdAgainstWhichToDetect === ChainId.mainnet;
-    const { tokensChainsCache } = this.messagingSystem.call(
-      'TokenListController:getState',
-    );
-    this.#tokensChainsCache = isTokenDetectionInactiveInMainnet
-      ? this.#getConvertedStaticMainnetTokenList()
-      : tokensChainsCache ?? {};
 
-    const tokenCandidateSlices = this.#getSlicesOfTokensToDetect({
+    // Execute the rest in a loop for each pair of chainId and networkClientId in the result array
```

Comment:
The accounts api is capable of querying multiple chains in a single http request, so we'd rather get them all in 1 go.  At least for the chains it supports.  We would still fall back to the RPC approach for unsupported chains

https://docs.cx.metamask.io/api/multichainAccounts/#tag/Balances/operation/AccountsController_getAccountBalancesV2